### ISSUE TITLE
adding a new language

### ISSUE THREAD

**Issue by tculjaga** [huggingface/parler-tts#128]
Hi, is it possible to add support for a new language as slovenian/croatian/serbian ?

Do you have any procedure we can follow to train the model for that languages ?

**Comment by ylacombe** [comment]
Hey @tculjaga, it's possible but there's no guide yet! I don't have the bandwidth for the next few weeks though

**Comment by Oyemade** [comment]
Would you mind sharing some pointers? I don't mind taking a stab at it. I just successfully fine-tuned MMS for some Western African languages and hoping to build off of that.

**Comment by tculjaga** [comment]
Hi @ylacombe thanks for a quick response. I'd like to give it a try... all we need is a small and short guide (a bullet list is enough for start) how you do it on a specific language then we can try our best to battle through it :)

**Comment by SherryS997** [comment]
> Would you mind sharing some pointers? I don't mind taking a stab at it. I just successfully fine-tuned MMS for some Western African languages and hoping to build off of that.

Hello, we're actually trying to train parler-tts for Indian languages and our pilot test gave great results. We followed exactly the same training process, and only changed to text encoder to mT5. It worked well!

**Comment by ylacombe** [comment]
Hey @SherryS997, 
thanks for giving this feedback. I actually trained a version of Mini v1 which uses mT5 instead of T5, but it's only in English. 
Do you think you'd be able to either open-source your model, or to share some snippet?
Thanks!

**Comment by SherryS997** [comment]
Hi @ylacombe,
Yes, we plan to open-source our model, along with the dataset, code, and captions, in the next month or two once we’re fully satisfied with the results. This work is part of the TTS research at AI4Bharat, and the model will be designed to support all 23 official languages of India, including English. We're also experimenting with various English accents and are optimistic that the model will handle those effectively as well.
In our pilot training, as stated before, we used mT5, which gave excellent results in terms of performance. However, we are transitioning to ai4bharat/IndicBERTv2-SS for the text encoder and tokenizer because mT5 only covers 12 of the 23 languages we aim to support. IndicBERTv2-SS offers broader language support, making it better suited to our goals.

**Comment by ylacombe** [comment]
Hey @SherryS997, let's speak over mail if that's okay with you: yoach [at] huggingface.co

**Comment by showgan** [comment]
Hi @ylacombe and @SherryS997,
I'm also very interested in training for a low resource language which is not supported by any tokenizer. I'd really appreciate it if you could share with me some advice as well.
Thanks!

**Comment by Strive-for-excellence** [comment]
> Hi @ylacombe, Yes, we plan to open-source our model, along with the dataset, code, and captions, in the next month or two once we’re fully satisfied with the results. This work is part of the TTS research at AI4Bharat, and the model will be designed to support all 23 official languages of India, including English. We're also experimenting with various English accents and are optimistic that the model will handle those effectively as well. In our pilot training, we used mT5, which delivered excellent results. However, we are now experimenting with other tokenizers, as mT5 supports only 12 of the 23 languages we’re targeting. These alternatives will help us achieve broader language coverage for the project.

Great job. Looking forward to your open-source release.

**Comment by SherryS997** [comment]
> Hi @ylacombe and @SherryS997, I'm also very interested in training for a low resource language which is not supported by any tokenizer. I'd really appreciate it if you could share with me some advice as well. Thanks!

You may need to build a tokenizer or better yet extend the flanT5 tokenizer. Then all you need to do is point to this tokenzer is the start json config.

**Comment by skbose** [comment]
Hey @SherryS997, I've been trying to fine-tune the parler-tts model for the NPTEL TTS dataset (basically Indian English). The loss doesn't seem to reduce [wandb link](https://wandb.ai/mrv31-microsoft/parler-speech/reports/train-loss-24-10-16-10-39-57---Vmlldzo5NzM2NTg2?accessToken=seizz6g3s2a3ghmg504lykoelhr670voovx0r3yspkkybxq56g0otcb0gtzz3o1d). I have tried training longer (>12K) steps as well with no luck. How do I debug the issue? Do you think modifying the tokeniser would help?

**Comment by SherryS997** [comment]
If it's just English, no need to change anything wrt the tokenizer. I'll suggest using the IndicTTS dataset for Indian English. It is a high quality TTS dataset unlike the NPTEL dataset. I'd also recommend naming a few speakers as named speakers seem to be the best at the moment in capturing accents, styles, voice quality, etc.

**Comment by imPdhar** [comment]
@ylacombe Just curious: How do I train parler for the arabic language?

**Comment by tculjaga** [comment]
Hi @ylacombe, did you have a chance to put a sort of "how-to" for adding a low resource language?

I'd really like to give it a try.

**Comment by SaleemMalikAI** [comment]
Hi  @ylacombe i just want to train that model for my local language and i search that the google/flan-t5-base is also supports the urdu language is it i still need to add an other tokenizer that supports the urdu language or just run the taining code
